Documentation/nvme-set-reg.txt

```diff
@@ -0,0 +1,102 @@
+nvme-set-reg(1)
+===============
+
+NAME
+----
+nvme-set-reg - Writes and shows the defined NVMe controller register
+
+SYNOPSIS
+--------
+[verse]
+'nvme set-reg' <device> [--offset=<offset>, -O <offset>]
+			[--value=<val> | -V <val>] [--mmio32 | -m]
+			[--intms=<val>] [--intmc=<val>] [--cc=<val>]
```

Comment:
After starring at the code for a while I think this scheme of listening every register by name makes it unnecessary complex.

I suggest we just have 

```
nvme set-reg <device> [--offset=<offset>, -O <offset>]
					[--value=<val> | -V <val>] [--mmio32 | -m]
					[--name=<str> | -n <str>]
```

and just map the name to the offset as first step and just use the offset in the code. The mixing of the offset and option names is too messy. And if you still want to have the `--csts` naming scheme just map it at the top level to the offset.

Reply:
The reason to have the `--csts` naming is to be able to specify multiple registers to set or get and easy to specify the argument by the tab completion so the code is a little bit complecated. If changed to use `--name` instead I think it is not able to specifify multiple registers and needed to check the register name to specify by the help or documentation.

Reply:
Fair enough. Though I still think you make it way to complex. If we introduce command line option for the named registers, just use this information on the top level and do something like

```c
if (argconfig_parse_seen(opts,` "intms"))
    `set_register(INTMS_OFFSET,` opts.intms);
```

Reply:
Fixed as mentioned I think so please review again.